builder/resubmit_utils.go

```diff
@@ -9,13 +9,28 @@ import (
 )
 
 // runResubmitLoop checks for update signal and calls submit respecting provided rate limiter and context
-func runResubmitLoop(ctx context.Context, limiter *rate.Limiter, updateSignal chan struct{}, submit func()) {
+func runResubmitLoop(ctx context.Context, limiter *rate.Limiter, updateSignal chan struct{}, submit func(), submitTime *time.Time) {
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-updateSignal:
-			res := limiter.Reserve()
+			var res *rate.Reservation
+			if submitTime == nil {
+				res = limiter.Reserve()
```

Comment:
since you reserve in the end of both branches of the if-else you can move it out and put it after

Reply:
good point, updated